packages/strapi-plugin-settings-manager/controllers/SettingsManager.js

```diff
@@ -0,0 +1,244 @@
+
+    const model = Service.databases(name, env);
+    const items = Service.getItems(model);
+
+    params = Service.cleanParams(params, items);
+
+    let validationErrors;
+    [params, validationErrors] = Service.paramsValidation(params, items);
+
+    if (!_.isEmpty(validationErrors)) return ctx.badData(null, Service.formatErrors(validationErrors));
+
+    Service.installDependency(params, name);
+
+    strapi.reload.isWatching = false;
+
+    const updateErrors = Service.updateSettings(params, items, env);
+
+    if (!_.isEmpty(updateErrors)) return ctx.badData(null, Service.formatErrors(updateErrors));
+
+    ctx.send({ ok: true });
+
+    strapi.reload();
+  },
+
+  updateDatabase: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { name, env } = ctx.params;
+    let params = ctx.request.body;
+
+    if (!env || _.isEmpty(_.find(Service.getEnvironments(), { name: env }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.environment.unknow' }] }]);
+    if (!name || _.isEmpty(_.find(Service.getDatabases(env), { name }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.database.unknow' }] }]);
+
+    const model = Service.databases(name, env);
+    let items = Service.getItems(model);
+
+    params = Service.cleanParams(params, items);
+
+    let validationErrors;
+    [params, validationErrors] = Service.paramsValidation(params, items);
+
+    if (!_.isEmpty(validationErrors)) return ctx.badData(null, Service.formatErrors(validationErrors));
+
+    const newName = _.get(params, `databases.connections.${name}.name`);
+
+    if (newName && newName !== name) {
+      params = _.assign(_.clone(strapi.config.environments[env].databases.connections[name]), params.databases.connections[name]);
+      delete params.name;
+
+      const connections = _.clone(strapi.config.environments[env].databases.connections);
+      connections[newName] = params;
+      connections[name] = undefined;
+
+      params = { databases: { connections }};
+
+      items = [{ target: 'databases.connections' }];
+    }
+
+    strapi.reload.isWatching = false;
+
+    Service.installDependency(params, newName);
+
+    const updateErrors = Service.updateSettings(params, items, env);
+
+    !_.isEmpty(updateErrors) ? ctx.badData(null, Service.formatErrors(updateErrors)) : ctx.send({ ok: true });
+
+    strapi.reload();
+  },
+
+  deleteDatabase: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { name, env } = ctx.params;
+
+    if (!env || _.isEmpty(_.find(Service.getEnvironments(), { name: env }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.environment.unknow' }] }]);
+    if (!name || _.isEmpty(_.find(Service.getDatabases(env), { name }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.database.unknow' }] }]);
+
+    const connections = _.clone(strapi.config.environments[env].databases.connections);
```

Comment:
Few questions about this method:

1. Why do you need to clone the object?
2. Why you don't restart the server after the deletion?
3. Do you also remove the hook (strapi-mongoose, strapi-bookshelf) from the `node_modules` if you don't need it anymore?
4. What happens to the models which are related to the deleted connection?